File: Selenium/Activities/Activity17.java
import java.util.List;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;

public class Activity17 {

	public static void main(String[] args) {
		WebDriver driver = new FirefoxDriver();
		
		driver.get("https://training-support.net/webelements/selects");
		System.out.println("Title of the page: " +driver.getTitle());
		
		WebElement multiSelectDropDown = driver.findElement(By.xpath("//select[contains(@class, 'h-80')]"));
		
		Select multiSelect = new Select(multiSelectDropDown);
		
		multiSelect.deselectByIndex(0);
		multiSelect.selectByVisibleText("HTML");
		multiSelect.selectByIndex(3);
		multiSelect.selectByIndex(4);
		multiSelect.selectByIndex(5);
		multiSelect.selectByValue("nodejs");
		multiSelect.deselectByIndex(4);
		
		List<WebElement> selectedOptions = multiSelect.getAllSelectedOptions();
        System.out.println("Selected options are: ");
        for (WebElement option : selectedOptions) {
            System.out.println(option.getText());
        }
		driver.quit();
	}
}

File: Selenium/Activities/Activity17.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.select import Select

with webdriver.Firefox() as driver:
    driver.get("https://training-support.net/webelements/selects")
    print(f"Title of the page: {driver.title}")

    drop_down = driver.find_element(By.XPATH, "//select[contains(@class, 'h-80')]")
		
    multi_select = Select(drop_down)
		
    multi_select.deselect_by_index(0)
    multi_select.select_by_visible_text("HTML")
    multi_select.select_by_index(3)
    multi_select.select_by_index(4)
    multi_select.select_by_index(5)
    multi_select.select_by_value("nodejs")
    multi_select.deselect_by_index(4)
		
    selectedOptions = multi_select.all_selected_options
    print("Selected options are: ")
    for option in selectedOptions:
        print(option.text)